photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java

```diff
@@ -197,8 +215,13 @@ public Optional<EstimatedRobotPose> update() {
         }
 
         PhotonPipelineResult cameraResult = camera.getLatestResult();
+        if (poseCacheTimestampSeconds != -1 && Math.abs(poseCacheTimestampSeconds - cameraResult.getTimestampSeconds()) < 1e-6) {
+            return cachedPose;
+        }
+        poseCacheTimestampSeconds = cameraResult.getTimestampSeconds();
         if (!cameraResult.hasTargets()) {
-            return Optional.empty();
+            cachedPose = Optional.empty();
+            return cachedPose;
```

Comment:
We should probably actually just return empty here? We should assume users are using *DrivePoseEstimators from wpilib IMO, in which case adding the same measurement multiple times is Wrong

Reply:
I'm not sure how this would lead to adding the same measurement multiple times? `cachedPose` is assigned to empty before being returned.